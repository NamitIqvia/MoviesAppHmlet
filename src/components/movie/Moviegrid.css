.movie{
    width: 300px;
    margin: 1rem;
    box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.3);
    overflow: hidden;
    background-color: #4f3d8f;
    color: white;
    position:relative;
}

.movie img{
    object-fit: cover;
    height:300px;
    width:100%;
}

.movie-info{
    display:flex;
    padding-bottom:1rem;
    padding-left:0.3rem;
    padding-right:0.3rem;
    justify-content: space-between;
}

.spacer{
    width:20px;
}

.movie-info span{
    margin:0;
    padding:0;
}

.movie-info p{
    margin:0;
    padding:0;
}

.movie-info{
    margin-bottom:20px;
}

.detailsButton{
    position:absolute;
    bottom:0;
    left:0;
    background-color:#a14bdb;
    border:none;
    padding:5px;
}