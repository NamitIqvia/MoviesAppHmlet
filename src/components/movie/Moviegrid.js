import React from 'react';
import './Moviegrid.css';
import MovieDetails from './MovieDetails';
//import { useHistory } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

const IMG_API = "https://image.tmdb.org/t/p/w1280";
const Moviegrid = ({data}) => {
   


    return (
        <div className="movie">
            <img src={data.poster_path ? IMG_API+data.poster_path : "https://images.unsplash.com/photo-1512070750041-b9479c107194?ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8bW92aWVzfGVufDB8fDB8fA%3D%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"} alt={data.title} />
            <div className="movie-info">
                <p>{data.title}</p>
                <span className="spacer"></span>
                <span>{data.vote_average}</span>
            </div>
              <Link data-testid="details-link" to={{pathname:"/details",
                       state:{ movieData:data}
                     }}><button className="detailsButton">Details</button></Link>
        </div>
    )
}

export default Moviegrid;
