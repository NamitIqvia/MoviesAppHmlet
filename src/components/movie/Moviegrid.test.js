import React from 'react';
//import { Router } from 'react-router-dom';
import { BrowserRouter, Route } from 'react-router-dom';
import  Moviegrid  from './Moviegrid';
import renderer from 'react-test-renderer';




it('should take a snapshot', () => {
    const data ={title:"Test Movie", vote_average:0}
    const tree = renderer.create(<BrowserRouter><Moviegrid data={data} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});