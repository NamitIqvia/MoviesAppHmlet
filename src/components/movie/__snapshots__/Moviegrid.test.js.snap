// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div
  className="movie"
>
  <img
    alt="Test Movie"
    src="https://images.unsplash.com/photo-1512070750041-b9479c107194?ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8bW92aWVzfGVufDB8fDB8fA%3D%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60"
  />
  <div
    className="movie-info"
  >
    <p>
      Test Movie
    </p>
    <span
      className="spacer"
    />
    <span>
      0
    </span>
  </div>
  <a
    data-testid="details-link"
    href="/details"
    onClick={[Function]}
  >
    <button
      className="detailsButton"
    >
      Details
    </button>
  </a>
</div>
`;
